ksql-api/src/main/java/io/confluent/ksql/api/server/InsertsBodyHandler.java

```diff
@@ -79,16 +89,35 @@ public void handleBodyBuffer(final Buffer buff) {
         return;
       }
       final JsonObject properties = args.getJsonObject("properties");
-      routingContext.request().endHandler(this::handleBodyEnd);
-      acksSubscriber = acks ? new AcksSubscriber(routingContext.response()) : null;
+      acksSubscriber = acks ? new AcksSubscriber(ctx, routingContext.response()) : null;
```

Comment:
there are still quite a few relevant comments on #4320 that I still see are relevant. Can you please go through them and make sure they are addressed or comment there dismissing them?

Reply:
I've addressed them now on the other PR